dbms/src/Encryption/RateLimiter.cpp

```diff
@@ -295,31 +295,48 @@ ReadLimiter::ReadLimiter(
     Int64 get_io_stat_period_us,
     UInt64 refill_period_ms_)
     : WriteLimiter(rate_limit_per_sec_, type_, refill_period_ms_)
-    , getIOStatistic(std::move(getIOStatistic_))
-    , last_stat_bytes(getIOStatistic())
+    , get_io_statistic(std::move(getIOStatistic_))
+    , last_stat_bytes(get_io_statistic())
     , last_stat_time(now())
     , log(&Poco::Logger::get("ReadLimiter"))
     , get_io_statistic_period_us(get_io_stat_period_us)
-{}
+{
+    auto auto_update_available_balance_worker = [&]() {
+        while (!stop)
+        {
+            TimePoint us = now();
+            // Not call getIOStatisctics() every time for performance.
+            // If the clock back, elapsed_us could be negative.
+            Int64 elapsed_us = std::chrono::duration_cast<std::chrono::microseconds>(us - last_stat_time).count();
+            if (get_io_statistic_period_us == 0 || elapsed_us >= get_io_statistic_period_us)
+            {
+                refreshAvailableBalance();
+                LOG_FMT_INFO(log, "-*- available_balance {} -*-", available_balance);
+            }
+        }
+    };
 
-Int64 ReadLimiter::getAvailableBalance()
+    auto_update_available_balance_thread = std::thread(auto_update_available_balance_worker);
```

Comment:
I think this thread can be global instead of per `ReadLimiter` to since we may have more than one `ReadLimiter` objects. The job of thread is periodically calculate the foreground and background read I/O bytes. Also, maybe we can reuse the 'auto tune' thread to do this job.